app/src/main/java/org/thunderdog/challegram/emoji/Emoji.java

```diff
@@ -662,6 +662,12 @@ public EmojiInfo getEmojiInfo (CharSequence codeCs, boolean allowRetry) {
         return getEmojiInfo(Character.toString(code.charAt(0)) + code.charAt(2));
       }
     }
+    if (info == null) {
```

Comment:
This ignores the `allowRetry` argument. When it's `false`, there should be no additional lookups.